### PR TITLE
Add extraction logic to mitele for videos not based in ooyala

### PR DESCRIPTION
**Pull request by DjMoren**
## Please follow the guide below

- You will be asked some questions, please read them **carefully** and answer honestly
- Put an `x` into all the boxes [ ] relevant to your *pull request* (like that [x])
- Use *Preview* tab to see how your *pull request* will actually look like

---

### Before submitting a *pull request* make sure you have:
- [x] At least skimmed through [adding new extractor tutorial](https://github.com/ytdl-org/youtube-dl#adding-support-for-a-new-site) and [youtube-dl coding conventions](https://github.com/ytdl-org/youtube-dl#youtube-dl-coding-conventions) sections
- [x] [Searched](https://github.com/ytdl-org/youtube-dl/search?q=is%3Apr&type=Issues) the bugtracker for similar pull requests
- [x] Checked the code with [flake8](https://pypi.python.org/pypi/flake8)

### In order to be accepted and merged into youtube-dl each piece of code must be in public domain or released under [Unlicense](http://unlicense.org/). Check one of the following options:
- [x] I am the original author of this code and I am willing to release it under [Unlicense](http://unlicense.org/)
- [ ] I am not the original author of this code but it is in public domain or released under [Unlicense](http://unlicense.org/) (provide reliable evidence)

### What is the purpose of your *pull request*?
- [x] Bug fix
- [ ] Improvement
- [ ] New extractor
- [ ] New feature

---

### Description of your *pull request* and other information

Mitele has done changes, and some of the videos doesn't work with the extractor logic that was present (I'm not sure if it really works with anyone). That said, I've added new logic for videos not based in ooyala. This really fixes [this issue](https://github.com/ytdl-org/youtube-dl/issues/25582). The _account-needed_ tag added to that issue is not correct, it's only needed if you try to enter from outside Spain or if the video belongs to _mitele PLUS_ (premium videos), and the videos posted in that issue can be downloaded correctly with posted fix without any account. That not happens with the videos posted in [another issue](https://github.com/ytdl-org/youtube-dl/issues/24624), where the first is unavailable and the other belongs to _mitele PLUS_, that is not supported for the moment by the extractor logic (I don't have an account).

### CODE REVIEW

File: youtube_dl/extractor/mitele.py [hunk 1]
```diff
@@ -72,13 +71,10 @@ def _real_extract(self, url):
         title = pre_player['title']
         video = pre_player['video']
         video_id = video['dataMediaId']
-        content = pre_player.get('content') or {}
-        info = content.get('info') or {}
+        content = pre_player.get('content', {})
+        info = content.get('info', {})
```
Comment (dstftw) [inline youtube_dl/extractor/mitele.py:75]: This is not equivalent, do not touch.

File: youtube_dl/extractor/mitele.py [hunk 2]
```diff
@@ -91,3 +87,37 @@ def _real_extract(self, url):
...
+            caronte = self._download_json(
+                services['caronte'], video_id, 'Downloading caronte JSON')
+            cerbero_data = {
+                'bbx': caronte['bbx'],
+                'gbx': gbx['gbx']
+            }
+            headers = {
+                'Content-Type': 'application/json;charset=UTF-8',
+                'Origin': 'https://www.mitele.es'
+            }
```
Comment (dstftw) [inline youtube_dl/extractor/mitele.py:112]: Inline.

File: youtube_dl/extractor/mitele.py [hunk 3]
```diff
@@ -91,3 +87,37 @@ def _real_extract(self, url):
...
+            cerbero = self._download_json(
+                caronte['cerbero'], video_id, 'Downloading cerbero JSON',
+                headers=headers,
+                data=json.dumps(cerbero_data).encode('utf-8'))
+            stream = '%s?%s' % (caronte['dls'][0]['stream'],
+                                cerbero['tokens']['1']['cdn'])
```
Comment (dstftw) [inline youtube_dl/extractor/mitele.py:118]: Do not encode query into URL but pass via appropriate `_extract_m3u8_formats` parameter.

File: youtube_dl/extractor/mitele.py (hunk 2 above)
Comment (DjMoren) [reply youtube_dl/extractor/mitele.py:112]: I made like this for improving readability from my side as I read in the documentation, but I understand your point of view.

File: youtube_dl/extractor/mitele.py (hunk 3 above)
Comment (DjMoren) [reply youtube_dl/extractor/mitele.py:118]: I've passed it through query parameter, but the result isn't very readable from my point of view. Anyway, if this is the way to go, just go ahead and if you expected something different just tell me.

**Comment by DjMoren** [comment]
Force pushed an update in my branch, let's see if this is the definitive!

**Comment by felisucoibi** [comment]
thanks a ot it works

**Comment by N0mbre** [comment]
@DjMoren

Es posible descargar estas peliculas que son de pago sin ser premium como anteriormente.
https://www.mitele.es/peliculas/

**Comment by felisucoibi** [comment]
> @DjMoren
> 
> Es posible descargar estas peliculas que son de pago sin ser premium como anteriormente.
> https://www.mitele.es/peliculas/

NO, ahora son de pago.

**Comment by N0mbre** [comment]
He conseguido los m3u8 de los videos de pago sin ser premium pero no permite descargarlos da error 403 ¿hay alguna forma de saltarse el error 403?

**Comment by felisucoibi** [comment]
> He conseguido los m3u8 de los videos de pago sin ser premium pero no permite descargarlos da error 403 ¿hay alguna forma de saltarse el error 403?

Pagando, esta protegido para que sólo alguien que pague pueda bajarlos. Pero si pagas puedes conseguir los m3u8

**Comment by bdovaz** [comment]
Any news on this?

**Comment by felisucoibi** [comment]
stoped working?

**Comment by bdovaz** [comment]
@DjMoren can you make the requested change in order to merge this PR? Thanks.

**Comment by felisucoibi** [comment]
is broken again dont merge

**Comment by DjMoren** [comment]
Can you @felisucoibi or @bdovaz be more precise when you comment things? I have checked that tests are passing and done a manual test right now for _Mi casa es la tuya T09x98_ and everything was correct. If you find an error, please **detail** it, and if anyone have time to solve it will do. I only PRed this because previously it wasn't working when I needed it, but I'm not the main supporter, you should understand it.

PS: I'm not going to add support for payed videos because I'm not going to pay for an account here.

**Comment by bdovaz** [comment]
I have simply said if you can merge this PR, nothing more.

I haven't tried it because I don't know how to test something that is not published, I only use the Windows version.

**Comment by DjMoren** [comment]
No, you said `@DjMoren can you make the requested change in order to merge this PR? Thanks.`. The changes were done when I commented `Force pushed an update in my branch, let's see if this is the definitive!`. The reviewer has to approve the PR, I have requested to re-review now.

**Comment by bdovaz** [comment]
> No, you said `@DjMoren can you make the requested change in order to merge this PR? Thanks.`. The changes were done when I commented `Force pushed an update in my branch, let's see if this is the definitive!`. The reviewer has to approve the PR, I have requested to re-review now.

I meant because of this on the bottom:

**_Changes requested
1 review requesting changes_**

**Comment by DarkZeros** [comment]
Oh damm, I didn't search and ended redoing it miself on my branch -> https://github.com/ytdl-org/youtube-dl/pull/26757
I guess it is a duplicate now

**Comment by felisucoibi** [comment]
New urls not wroking.

2020-10-23 Cuarto Milenio - /programas-tv/cuarto-milenio/temporada-15/programa-623-40_1008608575015/player/

**Comment by felisucoibi** [comment]
Is possible to make this work with session videos that needs login and a cookie?

**Pull request closed without merging**